bebraven/platform
# Use Rise360ModuleVersion for grading

### Description

**Task**: https://app.asana.com/0/1174274412967132/1199733407123005

### Motivation and Context

We use the `activity_id` for the Rise360 module to look up the DB record and compute the grade based on progress and mastery quiz correctness (stored as `Rise360ModuleInteraction`). 

I broke this when I added the ability to update existing modules with `Rise360ModuleVersion`. Because the `Rise360Module`'s underlying zipfile can change, so can its `activity_id`.

This means we need to grade based on `Rise360ModuleVersion`,

### Test

- Assign non-zero points the Canvas assignment because we use a floating point with percent sign, "XX.X%", when submitting the grades to Canvas. Otherwise, any % of 0 possible points, which will always assign grade value of 0
- Login to Canvas a student, go through a module (need to generation interactions with `new=true` to trigger grading to run), in this example, I did 8% of the module and answered 0 mastery questions:
- <img width="1792" alt="Screen Shot 2021-01-12 at 10 10 52 AM" src="https://user-images.githubusercontent.com/62911934/104354902-c45ed200-54be-11eb-886a-415326cb45ca.png">
- In **deva**, run `bundle exec rake grade:modules` to manually trigger grading
- Check Canvas grades and see score updated. The computed grade of 4% of total 100 points:
<img width="1792" alt="Screen Shot 2021-01-12 at 10 10 39 AM" src="https://user-images.githubusercontent.com/62911934/104354961-d80a3880-54be-11eb-98e4-cbf206f33d9b.png">

### Types of changes




### Checklist:
